homeassistant/core.py

```diff
@@ -891,12 +899,17 @@ def async_listen(
         @callback that returns a boolean value, determines if the
         listener callable should run.
 
+        If run_immediately is passed, the callback will be run
+        right away instead of using call_soon
```

Comment:
Should we add a suggestion when to use this?
```suggestion
        If run_immediately is passed, the callback will be run
        right away instead of using call_soon. Only use this if
        the callback results in scheduling another task.
```